### PR TITLE
common/hooks/do-extract/00-distfiles.sh: support deb files

### PR DESCRIPTION
**Pull request by classabbyamp**
#### Testing the changes
- I tested the changes in this PR: **YES** (tested extraction/installation works for all packages with debs, packages generated are sane and are the same as before, except where deliberately changed)

### CODE REVIEW

**Comment by classabbyamp** [comment]
had to update yandex-disk due to the old version's distfile no longer existing upstream. seems to work from my basic testing but I don't have an account

File: srcpkgs/brother-dcp197c-cupswrapper/template [hunk 1]
```diff
@@ -13,13 +13,8 @@ distfiles="https://download.brother.com/welcome/dlf005405/dcp197ccupswrapper-${v
...
 nopie=yes
 
-do_extract() {
-	ar x ${XBPS_SRCDISTDIR}/${pkgname}-${version}/dcp197ccupswrapper-${version}-1.i386.deb
-}
-
 do_install() {
-	mkdir -p ${DESTDIR}
-	bsdtar xzpvf data.tar.gz -C ${DESTDIR}
+	vcopy opt /
```
[the remark this answers is not in the document]
Comment (classabbyamp) [reply srcpkgs/brother-dcp197c-cupswrapper/template:17]: I prefer having the create_wrksrc so I don't have to `vmkdir /opt`

**Review by ahesford — APPROVED** [review]
Not thrilled about making it easier to make packages from debs, but it clearly addresses a need.

**Comment by classabbyamp** [comment]
yeah I don't want to encourage debs, but the manual way is gross and inconsistent and increasing support for it could maybe make xdeb less attractive of an option for out-of-tree trees

**Pull request merged**